src/persistQueryClient/tests/PersistQueryClientProvider.test.tsx

```diff
@@ -147,7 +147,7 @@ describe('PersistQueryClientProvider', () => {
 
       return (
         <div>
-          <h1>{state.data}</h1>
+          <h1>{String(state.data)}</h1>
```

Comment:
@artysidorenko I think the react 18 type upgrades have shown an issue with the types of `useQueries`: Here, the type of `state.data` is inferred to `string | Promise<string> | undefined`, which cannot be rendered.

I will create a separate issue for this with a failing test case, just to let you know :)

Reply:
follow-up issue: https://github.com/tannerlinsley/react-query/issues/3483